docs-markdown/src/controllers/image-controller.ts

```diff
@@ -503,3 +512,44 @@ export function imageCompletionProvider() {
     completionItems.push(new CompletionItem(`:::image type="complex" source="" alt-text="" loc-scope="":::`));
     return completionItems;
 }
+
+export async function applyLink() {
+    // get editor, its needed to apply the output to editor window.
+    const editor = window.activeTextEditor;
+    if (!editor) {
+        noActiveEditorMessage();
+        return;
+    }
+
+    // if user has not selected any text, then continue
+    const RE_LINK = /:::image\s+((source|type|alt-text|lightbox|border|loc-scope)="([a-zA-Z0-9_.\/ -]+)"\s*)+:::/gm;
+    const position = new Position(editor.selection.active.line, editor.selection.active.character);
+    // get the current editor position and check if user is inside :::image::: tags
+    const wordRange = editor.document.getWordRangeAtPosition(position, RE_LINK);
+    if (wordRange) {
+        const options: InputBoxOptions = {
+            placeHolder: "Enter URL",
+            validateInput: (urlInput) => urlInput.startsWith("http://") || urlInput.startsWith("https://") ? "" :
```

Comment:
Can the urlInput aslo be a relative Url? "../../link/in/another/document.md"

Reply:
Hi @bharney0 , as far as I know this should only support links to web pages but I'll follow-up to make sure.

Reply:
Hey @bharney0 , good call.  We should support relative paths as well.  I removed the url check.  Thanks